src/main/java/dk/cs/aau/huppaal/presentations/TagPresentation.java

```diff
@@ -212,6 +215,37 @@ private void initializeShape() {
                 textField.requestFocus(); // This needs to be done twice because of reasons
             }
 
+            //Handle constraints for guards, selects, synchronizations, and updates
+            if(getParent() instanceof NailPresentation){
+                if(getParent().localToParent(getBoundsInParent()).getCenterX() > getComponent().widthProperty().doubleValue() - textField.getWidth()) {
```

Comment:
It seems the two cases are very similar, I think there should be a function (or just extract the object that you work on)

somthing like: 
if (getParent() instanof NailPresentation) {
o = getParent() 
} else {
o = getParent.getParent()
}

if (o.localToParent(...
...

Reply:
Definitely, must have missed it as I ran through it